src/Equations/damaged-elastic/Kernels/Time.cpp

```diff
@@ -703,3 +692,763 @@ void seissol::kernels::Time::flopsTaylorExpansion(long long& nonZeroFlops,
+      FluxInterpolatedBodyZ[timeInterval][8 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS + q] =
+          -szzNodal[q] / data.material.local.rho;
+      FluxInterpolatedBodyZ[timeInterval][9 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS + q] = 0;
+    }
+  }
+}
+
+std::tuple<real, seissol::kernels::Time::Stresses, seissol::kernels::Time::Stresses>
+    seissol::kernels::Time::calculateDamageAndBreakageStresses(real mu0,
+                                                               real alpha,
+                                                               real gammaR,
+                                                               real xi0,
+                                                               real xi,
+                                                               real lambda0,
+                                                               real EspI,
+                                                               real EspII,
+                                                               real stressXX,
+                                                               real stressYY,
+                                                               real stressZZ,
+                                                               real stressXY,
+                                                               real stressYZ,
+                                                               real stressXZ,
+                                                               real aB0,
+                                                               real aB1,
+                                                               real aB2,
+                                                               real aB3) {
+  seissol::kernels::Time::Stresses sS, sB;
+  real muEff = mu0 - alpha * gammaR * xi0 - 0.5 * alpha * gammaR * xi;
+  sS.sxx = lambda0 * EspI - alpha * gammaR * std::sqrt(EspII) + 2 * muEff * stressXX;
+  sS.syy = lambda0 * EspI - alpha * gammaR * std::sqrt(EspII) + 2 * muEff * stressYY;
+  sS.szz = lambda0 * EspI - alpha * gammaR * std::sqrt(EspII) + 2 * muEff * stressZZ;
+  sS.sxy = 2 * muEff * stressXY;
+  sS.syz = 2 * muEff * stressYZ;
+  sS.sxz = 2 * muEff * stressXZ;
+  sB.sxx = (2.0 * aB2 + 3.0 * xi * aB3) * EspI + aB1 * std::sqrt(EspII) +
+           (2.0 * aB0 + aB1 * xi - aB3 * xi * xi * xi) * stressXX;
+  sB.syy = (2.0 * aB2 + 3.0 * xi * aB3) * EspI + aB1 * std::sqrt(EspII) +
+           (2.0 * aB0 + aB1 * xi - aB3 * xi * xi * xi) * stressYY;
+  sB.szz = (2.0 * aB2 + 3.0 * xi * aB3) * EspI + aB1 * std::sqrt(EspII) +
+           (2.0 * aB0 + aB1 * xi - aB3 * xi * xi * xi) * stressZZ;
+  sB.sxy = (2.0 * aB0 + aB1 * xi - aB3 * xi * xi * xi) * stressXY;
+  sB.syz = (2.0 * aB0 + aB1 * xi - aB3 * xi * xi * xi) * stressYZ;
+  sB.sxz = (2.0 * aB0 + aB1 * xi - aB3 * xi * xi * xi) * stressXZ;
+
+  return std::make_tuple(muEff, sS, sB);
+}
+
+void seissol::kernels::Time::calculateStressesFromDamageAndBreakageStresses(real* qStressDofsXX,
```

Comment:
Use some structs to combine some of these arguments. This reduces the number of arguments, makes the signature more comprehensible and is less error prone: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c1-organize-related-data-into-structures-structs-or-classes

Reply:
Done

Reply:
done